src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/actionrule/AbstractActionRuleUpdater.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.ons.ctp.response.collection.exercise.service.impl.actionrule;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.response.action.representation.ActionRuleDTO;
+import uk.gov.ons.ctp.response.collection.exercise.client.ActionSvcClient;
+
+public abstract class AbstractActionRuleUpdater implements ActionRuleUpdater {
+  private static final String MULTIPLE_ACTION_RULES_FOUND_EXCEPTION =
+      "Multiple %s Action Rules Found for Action Plan %s, remove all but %d before continuing";
+  private static final String NO_ACTION_RULES_FOUND_EXCEPTION =
+      "No %s Action Rules Found for Action Plan %s, please create one instead";
+  private static final String ACTION_PLAN_NOT_FOUND_EXCEPTION = "Action Plan %s not found";
+  protected ActionSvcClient actionSvcClient;
+
+  public AbstractActionRuleUpdater(final ActionSvcClient actionSvcClient) {
+    this.actionSvcClient = actionSvcClient;
+  }
+
+  protected List<ActionRuleDTO> getActionRulesForActionPlan(final UUID actionPlanId)
+      throws CTPException {
+    final List<ActionRuleDTO> actionRules =
+        actionSvcClient.getActionRulesForActionPlan(actionPlanId);
+
+    if (actionRules == null) {
+      throw new CTPException(
```

Comment:
the current client implementation will blow up with a NPE if the service returns 404, so the actionRules returned will never be null

Reply:
definitely going to get a HttpClientErrorException, not null - have checked and tested

Reply:
Catching HttpClientErrorException in getActionRulesForActionPlan not and returning null on 404s.